src/Data/Array.purs

```diff
@@ -987,15 +987,15 @@ groupBy op xs =
     STA.unsafeFreeze result
 
 -- | Group equal elements of an array into arrays, using the specified
--- | equivalence relation to determine equality.
+-- | comparison relation to determine equality.
 -- |
 -- | ```purescript
--- | groupAllBy (\a b -> odd a && odd b) [1, 3, 2, 4, 3, 3]
--- |    = [NonEmptyArray [1], NonEmptyArray [2], NonEmptyArray [3, 3, 3], NonEmptyArray [4]]
+-- | groupAllBy (\a b -> compare (odd a) (odd b)) [1, 3, 2, 4, 3, 3]
+-- |    = [NonEmptyArray [2, 4], NonEmptyArray [1, 3, 3, 3]]
 -- | ```
```

Comment:
I adapted the example to the one above, but I don't think it's a good example. Any ideas?

Reply:
What about using `comparing Down`?

```purs
groupAllBy (comparing Down) [1, 3, 2, 4, 3, 3]
  = [NonEmptyArray [4], NonEmptyArray [3, 3, 3], NonEmptyArray [2], NonEmptyArray [1]]
```